src/org/omegat/core/machinetranslators/YandexCloudTranslate.java

```diff
@@ -0,0 +1,245 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+ with fuzzy matching, translation memory, keyword search,
+ glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2010 Alex Buloichik, Didier Briel
+ 2011 Briac Pilpre, Alex Buloichik
+ 2013 Didier Briel
+ 2016 Aaron Madlon-Kay
+ 2020 Lev Abashkin
+ Home page: http://www.omegat.org/
+ Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.machinetranslators;
+
+import org.omegat.gui.exttrans.MTConfigDialog;
+import org.omegat.util.JsonParser;
+import org.omegat.util.Language;
+import org.omegat.util.Log;
+import org.omegat.util.OStrings;
+import org.omegat.util.Preferences;
+import org.omegat.util.WikiGet;
+
+import java.awt.Window;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+/**
+ * Support of Yandex.Cloud machine translation.
+ *
+ * @author Alex Buloichik (alex73mail@gmail.com)
+ * @author Didier Briel
+ * @author Briac Pilpre
+ * @author Aaron Madlon-Kay
+ * @author Lev Abashkin
+ *
+ * @see <a href="https://cloud.yandex.com/docs/translate/api-ref/Translation/">Translation API</a>
+ */
+public class YandexCloudTranslate extends BaseTranslate {
+
+    protected static final String PROPERTY_OATH_TOKEN = "yandex.cloud.oauth-token";
+    protected static final String PROPERTY_FOLDER_ID = "yandex.cloud.folder-id";
+
+    protected static final int MAX_TEXT_LENGTH = 10000;
+    protected static final int AIM_TOKEN_TTL_SECONDS = 3600; // Recommended value
+
+    protected static final String IAM_TOKEN_URL = "https://iam.api.cloud.yandex.net/iam/v1/tokens";
+    protected static final String TRANSLATE_URL = "https://translate.api.cloud.yandex.net/translate/v2/translate";
+
+    private String AIMErrorMessage = null;
```

Comment:
Dumb question: Is AIM different from IAM? Because I see both. And I see that `iam` occurs in strings that must be correct in order for this connector to function, but `AIM` only occurs in identifiers, so I am curious if it's not a typo.

Reply:
The question is not dumb at all and it's really a typo.